### PR TITLE
Add a testing environment for Cognito

### PR DESCRIPTION
**Pull request by tsibley**
See commit messages for details.

Spurred by https://github.com/nextstrain/nextstrain.org/pull/581#issuecomment-1260075701.

### Testing

- [x] `terraform -chdir=env/production plan` reports only minor expected changes
- [x] `terraform -chdir=env/testing apply` worked (resources now exist)
- [x] Tests pass locally
- [x] Lint free locally
- [x] Checks pass

### CODE REVIEW

**Comment by tsibley** [comment]
Since this PR is dev-facing only and is something where I want to minimize divergence from `master` over time, I'm going to merge it later today or tomorrow morning. I'd still welcome post-merge reviews!

File: docs/terraform.rst [hunk 1]
```diff
@@ -15,11 +15,31 @@ AWS resources like Cognito user pools.
...
+==============
+
+All ``terraform`` commands below expect to be run from within a directory
+containing a Terraform configuration (i.e. a set of one or more ``*.tf`` or
+``*.tf.json`` files).
+
+We have the following configurations:
+
+  - :file:`env/production`
+
```
Comment (victorlin) [inline docs/terraform.rst:28]: `env/testing` is missing from this list.

File: docs/terraform.rst [hunk 2]
```diff
@@ -15,11 +15,31 @@ AWS resources like Cognito user pools.
...
+     ``terraform -chdir=env/production plan``.
+
+
 Setup
 =====
 
-`Install Terraform <https://www.terraform.io/downloads>`__ and then, from the
-root of this repository, run::
+`Install Terraform <https://www.terraform.io/downloads>`__ and then, from a
+configuration directory (e.g. :file:`env/production`), run::
```
Comment (victorlin) [inline docs/terraform.rst:42]: Suggestion: Make all the examples reference `env/testing`; that seems safer to me.

Comment (tsibley) [reply docs/terraform.rst:42]: Good suggestion!

File: docs/terraform.rst (hunk 1 above)
Comment (tsibley) [reply docs/terraform.rst:28]: Thanks for the catch!

Comment (tsibley) [reply docs/terraform.rst:28]: https://github.com/nextstrain/nextstrain.org/pull/661

File: docs/terraform.rst (hunk 2 above)
Comment (tsibley) [reply docs/terraform.rst:42]: https://github.com/nextstrain/nextstrain.org/pull/661

**Pull request merged**